File: internal/parser/python/testdata/calls.py
import os
import json
from datetime import datetime

def helper():
    return 42

def process_data(data):
    # Same-file call
    result = helper()
    # Import-qualified call
    path = os.path.join("/tmp", "data")
    encoded = json.dumps(data)
    now = datetime.now()
    return result

class DataProcessor:
    def validate(self, data):
        return len(data) > 0

    def process(self, data):
        # self method call
        if self.validate(data):
            # Import call from method
            return json.loads(data)
        return None

File: internal/parser/golang/testdata/calls.go
package calls

import (
	"encoding/json"
	"fmt"
	"os"
)

func helper() int {
	return 42
}

func formatOutput(data string) string {
	return fmt.Sprintf("output: %s", data)
}

func processData(input string) {
	result := helper()
	output := formatOutput(input)
	encoded, _ := json.Marshal(result)
	fmt.Println(output, string(encoded))
	os.Exit(0)
}

type Processor struct{}

func (p *Processor) validate(data string) bool {
	return len(data) > 0
}

func (p *Processor) Process(data string) string {
	if p.validate(data) {
		return formatOutput(data)
	}
	return ""
}
